cdap-app-fabric/src/main/java/co/cask/cdap/scheduler/ScheduleNotificationSubscriberService.java

```diff
@@ -136,31 +136,31 @@ protected void shutDown() throws Exception {
 
     @Nullable
     @Override
-    protected String loadMessageId(StructuredTableContext context) {
-      return null;
+    protected String loadMessageId(StructuredTableContext context) throws IOException {
+      return getJobQueue(context).retrieveSubscriberState(getTopicId().getTopic());
     }
 
     @Override
-    protected void storeMessageId(StructuredTableContext context, String messageId) {
-      // Intentional NO-OP until CDAP-14848 is resolved
+    protected void storeMessageId(StructuredTableContext context, String messageId) throws IOException {
+      getJobQueue(context).persistSubscriberState(getTopicId().getTopic(), messageId);
     }
 
     @Nullable
     @Override
     protected String loadMessageId(DatasetContext datasetContext) {
-      return getJobQueue(datasetContext).retrieveSubscriberState(getTopicId().getTopic());
+      return null;
```

Comment:
On second thought, are there any usages of the loadMessage/storeMessage using datasetContext left? If not can we remove in all classes?

Reply:
Good point - there are no more usages. This can be removed now

Reply:
I notice that there are other places too that can now be refactored to remove the old transaction executor. But it will lead to lots of unrelated changes to this PR. Since it is a simple re-factoring, I'll leave it to be done when CDAP-14848 is handled

Reply:
sg, thanks